### PR TITLE
tree.md: add more useful examples

### PR DESCRIPTION
**Pull request by waldyrious**
and reorder entries by increasing complexity

### CODE REVIEW

File: pages/common/tree.md [hunk 1]
```diff
@@ -14,14 +14,14 @@
...
 `tree -a`
 
-- Print human readable size of files:
+- Print the tree without indentation lines, showing the full path instead:
 
-`tree -h`
+`tree -i -f`
 
-- Print the full path for each file:
+- Print the size of each node next to it, in human-readable format, with folders displaying their cumulative size (as in the `du` command):
```
Comment (agnivade) [inline pages/common/tree.md]: This feels like too many options in a single command. Wonder if we can cut down on one of them.

Comment (waldyrious) [inline pages/common/tree.md]: We could remove the --du option, but that makes the output less practical. I wonder if what makes it seem complex is the length of the description. Would something like this be better?

"Print the human-readable size of each node next to it (with folders displaying their cumulative size, as in the `du` command):"

**Comment by waldyrious** [comment]
The -P option needs the -a to match dotfiles, and --matchdirs to match "dotdirs" like ".git".

I wonder if we could add `tree -a -P '.git' --prune --matchdirs` to fill this gap. THoughts?

File: pages/common/tree.md [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
 
-`tree -f`
+`tree -s -h --du`
 
-- Print the tree without lines and indentation. Useful when used with -f:
+- Filter the tree using a wildcard (glob) pattern, showing only the hierarchies leading to the matching entries:
 
-`tree -i`
+`tree -P {{*.txt}} --prune`
```
Comment (waldyrious) [inline pages/common/tree.md]: The -P option needs `-a` to match dotfiles, and --matchdirs to match "dotdirs" like ".git".

I wonder if we could add `tree -a -P '.git' --prune --matchdirs` to fill this gap, building on the simpler `-P` example proposed here. Thoughts?

File: pages/common/tree.md (hunk 1 above)
Comment (agnivade) [inline pages/common/tree.md]: Its not the length of the description. Although a shorter description does have lesser cognitive load.

But yea .. I agree the 3 options really need to go together. In that case, your revised statement certainly looks better.

File: pages/common/tree.md (hunk 2 above)
Comment (agnivade) [inline pages/common/tree.md]: This feels like straying too far from the 'most common usage' motto that we follow. This example seems sufficiently complex enough to be a last example. IMO, we don't need anything more complicated than this.

Comment (waldyrious) [inline pages/common/tree.md]: Fair enough, you've got a good point. I'll keep this particular invocation in my personal notes, though :)

File: pages/common/tree.md (hunk 1 above)
Comment (waldyrious) [inline pages/common/tree.md]: Since you seem unsure still, I'll wait for someone else to chime in :)

Comment (agnivade) [inline pages/common/tree.md]: You got me there :sweat_smile:

Comment (ostera) [inline pages/common/tree.md]: I think we're good with `Print the full path for each file with human-readable size`. Anything other than that seems like over-explaining to me :'D

File: pages/common/tree.md (hunk 2 above)
Comment (ostera) [inline pages/common/tree.md]: Ah, this is a good one. I typically end up `tree -a | grep -v`-ing out the things I don't want to see.

I'd stick with the simple and just use `-P {{pattern}}` and let people figure out the rest.

`--prune` seems excessive. Why filter out empty directories in this general an example? They won't hurt if `--matchdirs` isn't set as they contain nothing to match on.

`--matchdirs` is probably relevant to this pattern matching sample, but makes the command more daunting to deal with at first.

In my book, `-a` means `anarchic`. But most likely you're not trying to look for things inside `.git`. If you are, and you know of this folder, you likely know it's a _hidden_ folder and that it'll be ignored by default by most commands that follow this convention. 

I'd stick with the simple.

File: pages/common/tree.md (hunk 1 above)
Comment (waldyrious) [inline pages/common/tree.md]: @ostera you probably missed that the actual command changed :) I reordered the commands to keep them sorted by complexity. Please check the full diff.

File: pages/common/tree.md (hunk 2 above)
Comment (waldyrious) [inline pages/common/tree.md]: Good points, I actually meant to add `--matchdirs` here. But we could make this just about finding files within the tree, and make a separate `-P` example with `--matchdirs` and `--prune`, which I think is pretty useful too. That way we have both options, and the complexity doesn't suddenly jump up :D

File: pages/common/tree.md (hunk 1 above)
Comment (ostera) [inline pages/common/tree.md]: @waldyrious oh my! You're right, completely missed the command change.

File: pages/common/tree.md (hunk 2 above)
Comment (waldyrious) [inline pages/common/tree.md]: @agnivade @ostera actually, the `--prune` option is not implemented in linux's `tree`. Per the [manpage](http://linux.die.net/man/1/tree):

> ### Bugs

> Tree does not prune "empty" directories when the -P and -I options are used. Tree prints directories as it comes to them, so cannot accumulate information on files and directories beneath the directory it is printing.

That page also shows that the `--du` option does not exist for the linux version.

So I guess I'll change this entry to be simply `tree -P {{*.txt}}` and the one before it to be `tree -s -h`, then move the page to linux/tree.md, and create a separate osx/tree.md for the behavior including `--prune` and `--du`. Sounds good?

Comment (agnivade) [inline pages/common/tree.md]: Huh .. I wonder why this is so .. Anyway, SGTM.

**Comment by waldyrious** [comment]
@ostera @agnivade PTAL

**Comment by agnivade** [comment]
LGTM overall. Are we sure, this command isn't there for SunOS ?

**Comment by ostera** [comment]
I have an idea for _extending_ the `common` page with platform-specific commands...I think we should talk about this before duplicating data.

**Comment by waldyrious** [comment]
@ostera that's literally our oldest open issue (#190). I fully trust that we could come up with a nice solution in a few sessions of intensive brainstorming and implementation, but considering that issue's history and the need to update lots of clients over which we have no direct control, forgive me for not holding my breath :) I'll help with moving that forward, but would very much like to have this merged as I need to look up the syntax for some of these `tree` options all the time :P

**Comment by waldyrious** [comment]
@agnivade, we (well, I) don't know. We can't really enforce that verification when most of our contributors don't have access to SunOS machines. Ideally we'd be able to set up different environments for CI, but from a cursory look Travis [only](https://docs.travis-ci.com/user/ci-environment/) offers [OS X and Linux (Ubuntu)](https://docs.travis-ci.com/user/multi-os/) images, and CircleCI [only has Linux (Ubuntu) ones](https://circleci.com/docs/build-images/).

That does give me an idea, though -- I wonder if we could set up automated tests for linux and osx, to verify that a command exists in a system (`type command`) and whether the command supports the options in the examples (something like `grep --foo > /dev/null 2>&1 || echo "fail"`?). But I'm not sure this could work in general. Just an idea :)

**Comment by ostera** [comment]
@waldyrious haha I had forgotten about that! Then ignore me for now.

**Comment by agnivade** [comment]
> We can't really enforce that verification when most of our contributors don't have access to SunOS machines.
I understand. No issues, lets move forward.

**Pull request merged**